serverlessworkflow/modules/ROOT/pages/cloud/build-and-deploy-with-serverless-operator-on-kubernetes.adoc

```diff
@@ -0,0 +1,178 @@
+
++
+[source,shell,subs="attributes+"]
+----
+kubectl get pod -n kogito-serverless-operator-system --watch
+----
+
+You can also follow the operator’s log:
+
++
+[source,shell,subs="attributes+"]
+----
+kubectl logs deployment/kogito-serverless-operator-controller-manager -n kogito-serverless-operator-system -f
+----
+====
+
+Once the operator is running, it will watch for new custom resources so that you can prepare your environment to be ready to build new {context} based on the definitions you will send to the operator.
+
+.Preparing for the build
+
+Follow these steps to create a container that you can deploy as a service on Kubernetes or KNative.
+
+* Create a namespace for the building phase
+
+Let's create a new namespace that will hold all the resources that we (or the operator) will create (pods, deployments, services, secretes, config map, and custom resources) in this guide.
+
+[source,bash,subs="attributes+"]
+----
+kubectl create namespace kogito-workflows
+----
+* Create a secret for the container registry authentication
+[source,bash,subs="attributes+"]
+----
+kubectl create secret docker-registry regcred --docker-server=<registry_url> --docker-username=<registry_username> --docker-password=<registry_password> --docker-email=<registry_email> -n kogito-workflows
+----
+or you directly import your local docker config into your kubernetes cluster:
+[source,bash,subs="attributes+"]
+----
+kubectl create secret generic regcred --from-file=.dockerconfigjson=${HOME}/.docker/config.json --type=kubernetes.io/dockerconfigjson -n kogito-workflows
+----
+* Create a platform containing the configuration (i.e. registry address, secret) for building your workflows
+
+The Platform CR is the resource used to control the behavior of the {product_name} {context} operator.
+
+It defines the behavior of all Custom Resources (Workflow and Build) in the given namespace.
+
+Since the {product_name} {context} operator is installed in global mode, you will need to specify a Platform CR in each namespace where you want the operator to be executed.
+
+You can find a basic Platform CR example in the config folder that you can simply apply to configure your operator.
+
+[source,bash,subs="attributes+"]
+----
+kubectl apply -f config/samples/sw.kogito_v1alpha08_kogitoserverlessplatform.yaml -n kogito-workflows
+Note: In this Custom Resource, spec.platform.registry.secret is the name of the secret you created just before.
+----
+
+[TIP]
+====
+You can also update "on-the-fly" the platform CR registry field with this command (change <YOUR_REGISTRY>)
+
+[source,bash,subs="attributes+"]
+----
+cat config/samples/sw.kogito_v1alpha08_kogitoserverlessplatform.yaml | sed "s|address: .*|address: <YOUR_REGISTRY>"
+----
+
+In order to retrieve the Cluster IP address of Minikube's internal registry use the following command:
+
+[source,bash,subs="attributes+"]
+----
+kubectl get svc registry -n kube-system -ojsonpath='{.spec.clusterIP}'
```

Comment:
What about using the external registry address ?
as  Minikube has been started with the insecure-registry, it is easier for the user to understand

Reply:
If you start minikube with the suggested parameters including the insecure-registry one and you check the output of the command:
`kubectl get service --namespace kube-system`
there isn't an external IP available for the registry.
In the Platform CR we need to put an IP address that could be reachable for the Kaniko pod because the push will be performed by a pod inside the cluster.
Isn't this the easiest way?

Reply:
When using the minikube registry addon, the registry is available externally on `$(minikube ip):5000` port
but internal IP will do indeed. Maybe set the explanation why so.